makeown.lua

```diff
@@ -76,8 +82,45 @@ local function fix_unit(unit)
 
     unit.civ_id = df.global.plotinfo.civ_id;
 
-    if  unit.profession == df.profession.MERCHANT then  unit.profession = df.profession.TRADER end
-    if unit.profession2 == df.profession.MERCHANT then unit.profession2 = df.profession.TRADER end
+    if convert_professions[unit.profession] then unit.profession = convert_professions[unit.profession] end
+    if convert_professions[unit.profession2] then unit.profession2 = convert_professions[unit.profession2] end
+end
+
+local function fix_army(unit)
+    -- Disassociate them from their army controller
+    if unit.enemy.army_controller then
+        if unit.enemy.army_controller.commander_hf == unit.hist_figure_id then
+            unit.enemy.army_controller.commander_hf = -1
+        end
+        unit.enemy.army_controller_id = -1
+        unit.enemy.army_controller = nil
+    end
+    fixUnitEnemyStatus(unit)
+end
+
+-- Also used by the script fix/loyaltycascade
+function fixUnitEnemyStatus(unit)
```

Comment:
"fix" is subjective. I think this should be named after what it actually does: `clearUnitEnemyStatus`